SpatialGDK/Source/Private/Interop/SpatialSender.cpp

```diff
@@ -91,26 +91,42 @@ Worker_RequestId USpatialSender::CreateEntity(USpatialActorChannel* Channel)
 	WriteAclMap ComponentWriteAcl;
 	ComponentWriteAcl.Add(SpatialConstants::POSITION_COMPONENT_ID, ServersOnly);
 	ComponentWriteAcl.Add(SpatialConstants::ROTATION_COMPONENT_ID, ServersOnly);
-	ComponentWriteAcl.Add(Info->SingleClientComponent, ServersOnly);
-	ComponentWriteAcl.Add(Info->MultiClientComponent, ServersOnly);
-	ComponentWriteAcl.Add(Info->HandoverComponent, ServersOnly);
-	ComponentWriteAcl.Add(Info->RPCComponents[RPC_Client], OwningClientOnly);
-	ComponentWriteAcl.Add(Info->RPCComponents[RPC_Server], ServersOnly);
-	ComponentWriteAcl.Add(Info->RPCComponents[RPC_CrossServer], ServersOnly);
-	ComponentWriteAcl.Add(Info->RPCComponents[RPC_NetMulticast], ServersOnly);
 
-	for (UClass* SubobjectClass : Info->SubobjectClasses)
+	ForAllSchemaComponentTypes([&](EComponentType Type) {
+		Worker_ComponentId ComponentId = Info->SchemaComponents[Type];
+		if (ComponentId == 0)
+		{
+			return;
+		}
+
+		if (Type == TYPE_ClientRPC)
+		{
+			ComponentWriteAcl.Add(ComponentId, OwningClientOnly);
```

Comment:
Another nit: Can we have a single call to Add, and the type define a local variable for the worker set.